### ISSUE TITLE
`id-denylist` breaks ESLint per Documentation

### ISSUE THREAD

**Issue by brenr** [eslint/eslint#13534]
Currently, the documentation is ambiguous and should use a different syntax. When a user unknowingly removes the first `"error"` it causes an unexpected breakage.

Change the structure from this:
```
"id-denylist": ["error", "data", "err", "e", "cb", "callback"]
```

To something like this:
```
"id-denylist": ["error", ["data", "err", "e", "cb", "callback"]]
```

If not do-able, have it implicitly assigned as "error" or "warn" if neither is specified as the first argument.

Currently the only way to diagnose this issue is by linting via CLI. It goes unnoticed in *Problems* in VSCode.

**Comment by anikethsaha** [comment]
Thanks for the Issue. 

Though in the docs, if you see the `options`, for `incorrect` and `correct` , 

> Examples of correct code for this rule with sample `"data", "callback"` restricted identifiers:
> ```js
> /*eslint id-denylist: ["error", "data", "callback"] */
> 
> var encodingOptions = {...};
> 
> function processFileResult() {
>     // ...
> }
> ```

With this, I guess it is quite clear. 
Maybe we can update the docs by breaking down the options and explaining the options?

**Comment by mdjermanovic** [comment]
This was discussed in #11331 for another rule with a similar configuration, but it seems that we didn't open a follow-up issue to enhance the rule page template.

**Comment by aladdin-add** [comment]
I prefer `"id-denylist": ["error", ["data", "err", "e", "cb", "callback"]]` -- it is more clear.

For similar rules, it would be great to deprecate the old-style rule options, and might remove them in next major release. However, an RFC is required to do so!(as eslint does not have a rule option deprecation policy atm.)

**Comment by brenr** [comment]
I agree - maybe something like along the lines of this:
```
{
    "id-enforce": [ "error", {
        "restrict": [ // These will tell you it's too ambiguous and you must choose another name
            "data",
            "callback",
            "cb",
            "handler"
        ],
        "prefer": { // This could provide a quick fix if a 1:1 mapping is found.
            "error": [ "e", "err" ], // Either of these can be quick fixed to rename to "error"
            "result": "res", // This is possibility #1 for either result or response
            "response": "res", // This is possibility #2 for either result or response
            "request": "req" // Like "error", this can be quick fixed.
        }
    } ]
}
```

**Comment by mdjermanovic** [comment]
> I agree - maybe something like along the lines of this:
> 
> ```
> {
>     "id-enforce": [ "error", {
>         "restrict": [ // These will tell you it's too ambiguous and you must choose another name
>             "data",
>             "callback",
>             "cb",
>             "handler"
>         ],
>         "prefer": { // match: replacement(s). 1:1 mappings can be quick fixed.
>             "e(rr)?": "error",
>             "res": [ "result", "response" ],
>             "req": "request"
>         }
>     } ]
> }
> ```

This looks like a proposal for a new rule, maybe we should focus on how to improve the docs (or configuration schema) for the existing one.
